app/retail/templates/shared/head.html

```diff
@@ -63,6 +63,9 @@
 <title>{% if title %}{{title}}{% else %}{% trans "Grow Open Source" %}{% endif %} | Gitcoin</title>
 {% include 'shared/favicon.html' %}
```

Comment:
from the metamask blog post about preparing your dapp for EIP 1102

```
Previously, dapps would access the MetaMask provider by using window.web3.currentProvider. While this will still work, the new, standard way to access the Ethereum provider in a Web browser is to use window.ethereum.
```

I dont see any updates for this in this repo...

Reply:
Oh, I completely missed this. Will make the updates.

Reply:
@owocki where is web3.currentProvider used and for what purpose?

Reply:
```
Searching 18920 files for "currentProvider"

/Users/kevinowocki/Sites/gitcoin/web/app/assets/ens/index.js:
    4    var method = 'personal_sign';
    5  
    6:   web3.currentProvider.sendAsync({
    7      method,
    8      params,

/Users/kevinowocki/Sites/gitcoin/web/app/assets/onepager/js/web3alert.js:
   25  };
   26  var metaMaskWarning = function() {
   27:   if (typeof web3 == 'undefined' || !web3.currentProvider || !web3.currentProvider.isMetaMask) {
   28      if (typeof document.suppressweb3alert != 'undefined') {
   29        _alert({ message: gettext('You must install <a href=https://metamask.io/>Metamask</a> to use this tool.') }, 'info');

/Users/kevinowocki/Sites/gitcoin/web/app/assets/v2/js/truncate-hash.js:
   70  new metamaskAddress();
   71  
   72: web3.currentProvider.publicConfigStore.on('update', function(e) {
   73    new metamaskAddress();
   74  });

/Users/kevinowocki/Sites/gitcoin/web/app/assets/v2/js/pages/kudos_send.js:
  561  }
  562  
  563: // web3.currentProvider.publicConfigStore.on('update', function(e) {
  564  var error;
  565  
  ...
  572    var network = e ? e.networkVersion : web3.version.network;
  573  
  574:   console.log(web3.currentProvider);
  575    if (network === '4' || network === '1') {
  576      console.log(network);

/Users/kevinowocki/Sites/gitcoin/web/app/assets/v2/js/pages/wallet_estimate.js:
    1  window.addEventListener('load', function() {
    2:   if (web3.currentProvider.isTrust) {
    3      $('#trust_label').show();
    4    } else {


```

You can find the answer by grepping the codebase.  I dont know more about the 'why' than you do.

Reply:
@owocki From my discussions with @PumpkingWok, It is most probably updated only in the below case:
```javascript
if (window.ethereum) {
    window.web3 = new constants.Web3(window.ethereum) // updated case
    try {
      await window.ethereum.enable()
    } catch (error) {
      console.error(error)
      console.error(`Refresh the page to approve/reject again`)
      window.web3 = null
    }
  } else if(window.web3) { // for legacy dapp browsers
    window.web3 = new constants.Web3(window.web3.currentProvider)
  }
```